add InstallmentPlan endpoint

fricha05 opened the pull request:
Adds the installment plan endpoint (get, create, update)

frichard-crypto committed:
add InstallmentPlan class, resource and tests

Dixeat commented:
Up

pierre-pg commented on payplug/resources.py:
```diff
@@ -391,3 +391,32 @@ class Operation(APIResource):
         pass
+
+
+class InstallmentPlan(APIResource):
```
Je pense qu'on voudrait : 

```suggestion
class InstallmentPlan(APIResource, VerifiableAPIResource, ReconstituableAPIResource):
```

Comme pour les autres ressources d'api.

pierre-pg commented on payplug/resources.py:
```diff
@@ -391,3 +391,32 @@ class Operation(APIResource):
+    """
+    object_type = 'intallment_plan'
+
+    """
+    Create an InstallmentPlan
+
+    :param data: the data to create the installment plan
+    """
+    def create(self, **data):
+        return payplug.InstallmentPlans.create_installment_plan_endpoint(data)
+
+    """
+    Update an InstallmentPlan
+
+    :param data: the data needed to update the installment plan
+    """
+    def update(self, **data):
+        return payplug.InstallmentPlans.update_installment_plan_endpoint(self, data)
+
+    """
+    Get an InstallmentPlan
+    """
+    def get(self):
+        return payplug.InstallmentPlans.get_installment_plan_endpoint(self)
```
Sauf erreur de ma part, les `payplug.InstallmentPlans.xxx` appelés ici n'existent pas.

Par ailleurs, je pense qu'ici on a seulement besoin d'une méthode `get_consistent_resource` (et peut-être un `update`, à voir avec les autres ressources).

fricha05 replied:
Ok je vois ce qu'il fallait faire, j'avais mal compris le principe de consistent resource et je m'étais trompé sur la manière de l'implémenter. Le prochain commit devrait corriger ça !

pierre-pg replied:
Merci !

As-tu testé ces ajouts de bout-en-bout ? (Avec de véritables calls api sur notre environnement de prod ou de QA.) Et si non, peux-tu effectuer de tels tests ? Merci !

fricha05 replied:
Yes, l'appel /v1/installment_plans/{installment_plan_id} renvoie bien le DAO InstallmentPlan quand il est appelé (testé en QA)

pierre-pg replied:
Super merci !

pierre-pg replied:
J'ai l'impression que cette PR n'ajoute que le GET... je me trompe ?

Il faudrait pouvoir faire tout ce qui est listé ici https://docs.payplug.com/api/apiref.html#installment-plans- (et gérer les notifications).

frichard-crypto committed:
fix consistent resource

Dixeat commented:
UP plz.

calops commented on payplug/__init__.py:
```diff
@@ -422,40 +422,18 @@ class InstallmentPlan:
-
-        http_client = HttpClient()
-        response, _ = http_client.patch(routes.url(routes.INSTALLMENT_PLANS, resource_id=installment_plan), data)
-        return resources.InstallmentPlan(**response)
```
Why remove these two methods? Any reason for only keeping the `GET` route?

fricha05 replied:
A misunderstanding on my part from a previous comment, the next commit should fix it

frichard-crypto committed:
add abort and create methods and tests

PhilDL committed:
Merge remote-tracking branch 'origin/installement-plans-endpoint' into feature-installment-plans

fricha05 commented:
UP plz

calops commented on payplug/test/test_init/test_dao_installment_plans.py:
```diff
@@ -0,0 +1,33 @@
+
+    def test_update(self):
+        installment_plan = payplug.InstallmentPlan.retrieve('installment_plan_id')
+        installment_plan = payplug.InstallmentPlan.abort(installment_plan)
```
Est-ce que ce test fonctionne ? D'après la méthode déclarée dans `__init__.py`, on attend un `installment_plan_id`, et non pas un `InstallmentPlan`.

calops replied:
@fricha05 petit up :slightly_smiling_face:

fricha05 replied:
Sry du délai (tendance à pas voir passé les notifs sur mon adresse perso).
Bien vu, le test fonctionne mais c'est quand même une typo, corrigée avec le dernier commit, merci :+1: !

PhilDL committed:
[IMP] InstallmentPlans api resource 

- Fix object name "intallment_plan"
- Refactor mapper InstallmentPlan data to have APIResource object like Payment
- Adding Schedule APIResource and initializer to map list to list of Schedule(s)
- Add tests notifications treat
- Fix tests to use new APIResource objects
- Bump version

calops committed:
Merge pull request #15 from SOGEXIS/feature-installment-plans

Finishing Feature installment plans

frichard-crypto committed:
typo
